### DISCUSSION TITLE
Blazor 需要 MVVM吗？

### DISCUSSION THREAD

**Discussion by TruePluto** [dotnetcore/BootstrapBlazor#1430] [Q&A] [answered]
在Blazor项目中的Counter.razor文件里
`private int currentCount = 0;`
当按钮点击是页面自动更新了`<p role="status">Current count: @CurrentCount</p>`
似乎不需要显式的通知页面，不像MVVM里面需要实现INotifyPropertyChanged
那么是如何实现的呢？ 在blazor开发中需要不需要MVVM呢？

**Comment by ArgoZhang** [comment, marked as answer]
@TruePluto `Blazor` 本身就是 `MVVM` 模式，微软也推荐这种模式，但是也支持传统的模式 `ref` 拿到组件实例，开发中尽量使用 `MVVM` 模式

这个模式主要是通过 `EventCallback` 来实现的，下面我给你一些参考资料

https://learn.microsoft.com/en-us/aspnet/core/blazor/components/event-handling?WT.mc_id=DT-MVP-5004174&view=aspnetcore-7.0#eventcallback

https://www.bilibili.com/video/BV1ZM411J7Z7/?spm_id_from=333.999.0.0&vd_source=d2d36c90ff4b675b5c3e044432a65bec

这个系列是我录制的免费视频，建议从头看一下。对 Blazor 会有一个比较完整清晰的认识

**Comment by densen2014** [comment]
几乎完全不需要显式的通知页面, 因为点击后会重新解析执行页面, 页面逻辑处理好基本就好了.
